app/src/main/java/com/android/sample/ui/settings/SettingsScreen.kt

```diff
@@ -0,0 +1,108 @@
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.sample.ui.navigation.NavigationActions
+import com.google.firebase.auth.FirebaseAuth
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SettingsScreen(navigationActions: NavigationActions) {
+  val context = LocalContext.current
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = { Text("Settings", fontSize = 20.sp) },
+            navigationIcon = {
+              IconButton(onClick = { navigationActions.goBack() }) {
+                Icon(imageVector = Icons.Outlined.ArrowBack, contentDescription = "Back")
+              }
+            },
+            modifier = Modifier.testTag("settingsScreen") // Add testTag for the screen itself
+            )
+      },
+      content = { paddingValues ->
+        Column(
+            modifier = Modifier.fillMaxSize().padding(paddingValues).padding(horizontal = 24.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              // User data button
+              Button(
+                  onClick = { /* TODO: Handle User Data */},
+                  modifier =
+                      Modifier.fillMaxWidth()
+                          .height(60.dp)
+                          .background(Color.LightGray, RoundedCornerShape(10.dp))
+                          .testTag("userDataButton"), // Test tag for User Data button
+                  colors = ButtonDefaults.buttonColors(containerColor = Color.LightGray)) {
+                    Text("User data", color = Color.Black)
+                  }
+
+              // Preferences button
+              Button(
+                  onClick = { /* TODO: Handle Preferences */},
+                  modifier =
+                      Modifier.fillMaxWidth()
+                          .height(60.dp)
+                          .background(Color.LightGray, RoundedCornerShape(10.dp))
+                          .testTag("preferencesButton"), // Test tag for Preferences button
+                  colors = ButtonDefaults.buttonColors(containerColor = Color.LightGray)) {
+                    Text("Preferences", color = Color.Black)
+                  }
+
+              Spacer(modifier = Modifier.height(200.dp)) // Adjust this height as needed
+
+              // Delete account button
+              Button(
+                  onClick = { /* TODO: Handle Account Deletion */},
+                  modifier =
+                      Modifier.fillMaxWidth()
+                          .height(48.dp)
+                          .testTag("deleteAccountButton"), // Test tag for Delete Account button
+                  colors = ButtonDefaults.buttonColors(containerColor = Color.Red)) {
+                    Icon(
+                        imageVector = Icons.Outlined.Delete,
```

Comment:
Suggestion: I think having individual composables such as DeleteButton or LogoutButton would make the code more readable.